src/commonMain/kotlin/com/charleskorn/kaml/YamlConfiguration.kt

```diff
@@ -83,12 +85,18 @@ public enum class SingleLineStringStyle {
     DoubleQuoted,
     SingleQuoted,
     Plain,
-    ;
+    PlainExceptAmbiguous;
```

Comment:
Might be good to add a KDoc comment here to explain what 'ambiguous' means here.